### ISSUE TITLE
ServerWithAnnotations doesn't do anything

### ISSUE THREAD

**Issue by joakime** [eclipse/jetty.project#3755]
The existing `ServerWithAnnotations` isn't handling its setup properly.

It sets a jndi variable called `maxAmount` on the webapp context.

https://github.com/eclipse/jetty.project/blob/dd79b516f33340b28c7d58742e8611281dab24c7/examples/embedded/src/main/java/org/eclipse/jetty/embedded/ServerWithAnnotations.java#L66-L67

This jndi entry is not used by any component in the declared webapp `test.war`

https://github.com/eclipse/jetty.project/blob/dd79b516f33340b28c7d58742e8611281dab24c7/examples/embedded/src/main/java/org/eclipse/jetty/embedded/ServerWithAnnotations.java#L54-L56

I suspect this is a typo, and we were meant to be using the `test-spec.war` instead, as there is a `@Resource` present that expects the `maxAmount` value.

https://github.com/eclipse/jetty.project/blob/dd79b516f33340b28c7d58742e8611281dab24c7/tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/AsyncListenerServlet.java#L40-L41

But this introduces a different problem, as the `new EnvEntry()` for `maxAmount` in the `ServerWithAnnotations` is overwritten by the `jetty-env.xml` when the `test-spec.war` is loaded.

https://github.com/eclipse/jetty.project/blob/dd79b516f33340b28c7d58742e8611281dab24c7/tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/webapp/WEB-INF/jetty-env.xml#L8-L13

In conclusion, we have a `ServerWithAnnotations` that isn't using the correct war file, and the example setting of `maxAmount` seems to have no impact on the end result because that setting isn't being honored.
And a question, should the `overrideWebXml` parameter on `new EnvEntry()` be able to override the `jetty-env.xml` usages?

**Comment by janbartel** [comment]
@joakime the webapp should be the test-spec.war.  There is nothing wrong with the Env-Entry for maxAmount, it is deliberately setting a value that is in turn overridden in jetty-env.xml to test overriding - the  webapp specifically checks that the overriding has happened, so please don't change it.

It may be the case that we shouldn't use the test-spec.war for this embedded example, as that webapp is doing some very specific tests and has some very specific setup. If that is confusing, perhaps we need a much simpler example webapp.

**Comment by gregw** [comment]
I like using the test-spec.war.   We just need some comments to explain what is going on.... and perhaps a use the JDBC in the webapp.

**Comment by janbartel** [comment]
@gregw what do you mean "use the JDBC in the webapp'? We provide a trivial implementation of Datasource (com.acme.MockDataSource) that the webapp has injected into it, and the webapp tests that the injection of the requested DataSource occurred and is not null. Why would we want MockDataSource to have to actually implement all the jdbc methods??
